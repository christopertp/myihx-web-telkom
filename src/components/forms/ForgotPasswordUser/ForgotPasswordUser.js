import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { Field } from 'redux-form';
import Button from '../../elements/Button';
import Spinner from '../../elements/Spinner';
import TextField from '../../fields/Text';
import styles from './styles.scoped.css';

export default function ForgotPasswordUser(props) {
  const { handleSubmit, invalid } = props;
  const { isLoading } = useSelector(s => s.forgotPassword);
  const inputProps = {
    label: 'MOBILE NUMBER OR EMAIL',
    placeholder: 'Enter your mobile number or email',
  };

  return (
    <form className={styles.root} onSubmit={handleSubmit}>
      <Field component={TextField} inputProps={inputProps} name="user" />
      <Button disabled={invalid || isLoading} type="submit">
        {isLoading ? <Spinner /> : 'Send Recovery Code'}
      </Button>
    </form>
  );
}

ForgotPasswordUser.defaultProps = {
  handleSubmit: () => {},
  invalid: false,
};

ForgotPasswordUser.propTypes = {
  handleSubmit: PropTypes.func,
  invalid: PropTypes.bool,
};
