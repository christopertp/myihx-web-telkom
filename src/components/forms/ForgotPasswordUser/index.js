import { reduxForm } from 'redux-form';
import ForgotPasswordUser from './ForgotPasswordUser';
import validate from './validate';

export default reduxForm({
  initialValues: {
    user: '',
  },
  form: 'forgotPasswordUser',
  validate,
})(ForgotPasswordUser);
