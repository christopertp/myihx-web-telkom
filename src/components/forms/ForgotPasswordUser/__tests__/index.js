import React from 'react';
import { useSelector } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import Spinner from '../../../elements/Spinner';
import ForgotPasswordUser from '../ForgotPasswordUser';

useSelector.mockImplementation(fn => {
  fn(({ login: '' }));
  return { isLoading: false, message: '' };
});

describe('src/components/forms/ForgotPasswordUser', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ForgotPasswordUser />);
    expect(tree).toMatchSnapshot();

    useSelector.mockImplementationOnce(() => ({ isLoading: true, message: 'error' }));
    ForgotPasswordUser.defaultProps.handleSubmit();
    const result = ForgotPasswordUser({
      ...ForgotPasswordUser.defaultProps,
    });
    const button = result.props.children[1];
    expect(button.props.children.type).toBe(Spinner);
  });
});
