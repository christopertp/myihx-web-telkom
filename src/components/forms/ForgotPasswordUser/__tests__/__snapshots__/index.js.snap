// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/forms/ForgotPasswordUser render 1`] = `
<form
  onSubmit={[Function]}
>
  <Field
    component={[Function]}
    inputProps={
      Object {
        "label": "MOBILE NUMBER OR EMAIL",
        "placeholder": "Enter your mobile number or email",
      }
    }
    name="user"
  />
  <Button
    className=""
    disabled={false}
    onClick={null}
    to=""
    type="submit"
    variant="red"
  >
    Send Recovery Code
  </Button>
</form>
`;
